Index: app/src/main/java/BroadcastReceiver/AlarmReceiver.java
===================================================================
--- app/src/main/java/BroadcastReceiver/AlarmReceiver.java	(revision 7332953f153c5c921c6b97532e8141a8e14b3ad5)
+++ app/src/main/java/BroadcastReceiver/AlarmReceiver.java	(revision 7332953f153c5c921c6b97532e8141a8e14b3ad5)
@@ -1,13 +0,0 @@
-package BroadcastReceiver;
-
-import android.content.BroadcastReceiver;
-import android.content.Context;
-import android.content.Intent;
-
-public class AlarmReceiver extends BroadcastReceiver {
-
-    @Override
-    public void onReceive(Context context, Intent intent) {
-
-    }
-}
Index: app/src/main/java/hanas/dnidomatury/AlarmNotificationReceiver.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/hanas/dnidomatury/AlarmNotificationReceiver.java	(date 1535488336643)
+++ app/src/main/java/hanas/dnidomatury/AlarmNotificationReceiver.java	(date 1535488336643)
@@ -0,0 +1,41 @@
+package hanas.dnidomatury;
+
+import android.app.Notification;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.support.v4.app.NotificationCompat;
+
+import hanas.dnidomatury.MainActivity;
+import hanas.dnidomatury.R;
+
+import static android.app.PendingIntent.FLAG_ONE_SHOT;
+
+public class AlarmNotificationReceiver extends BroadcastReceiver {
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        NotificationCompat.Builder builder = new NotificationCompat.Builder(context);
+
+        Intent myIntent = new Intent(context, MainActivity.class);
+        PendingIntent pendingIntent = PendingIntent.getActivity(
+                context,
+                0,
+                myIntent,
+                FLAG_ONE_SHOT );
+
+        builder.setAutoCancel(true)
+                .setDefaults(Notification.DEFAULT_ALL)
+                .setWhen(System.currentTimeMillis())
+                .setSmallIcon(R.mipmap.ic_launcher)
+                .setContentTitle("Zodiac")
+                .setContentIntent(pendingIntent)
+                .setContentText("Check out your horoscope")
+                .setDefaults(Notification.DEFAULT_LIGHTS | Notification.DEFAULT_SOUND)
+                .setContentInfo("Info");
+
+        NotificationManager notificationManager = (NotificationManager)context.getSystemService(Context.NOTIFICATION_SERVICE);
+        notificationManager.notify(1,builder.build());
+    }
+}
Index: app/src/main/java/hanas/dnidomatury/BootReceiver.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/hanas/dnidomatury/BootReceiver.java	(date 1535488336654)
+++ app/src/main/java/hanas/dnidomatury/BootReceiver.java	(date 1535488336654)
@@ -0,0 +1,27 @@
+package hanas.dnidomatury;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.widget.Toast;
+
+/**
+ * Created by azem on 11/3/17.
+ */
+
+public class BootReceiver extends BroadcastReceiver
+{
+
+    public void onReceive(Context context, Intent intent)
+    {
+
+        // Your code to execute when Boot Completd
+
+        Intent i = new Intent(context,MyService.class);

+        context.startService(i);
+
+        Toast.makeText(context, "Booting Completed", Toast.LENGTH_LONG).show();
+
+    }
+
+}
\ No newline at end of file
Index: app/src/main/java/hanas/dnidomatury/MainActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/hanas/dnidomatury/MainActivity.java	(revision 7332953f153c5c921c6b97532e8141a8e14b3ad5)
+++ app/src/main/java/hanas/dnidomatury/MainActivity.java	(date 1535489240361)
@@ -1,14 +1,16 @@
 package hanas.dnidomatury;
 
+import android.app.AlarmManager;
+import android.app.PendingIntent;
+import android.content.Context;
 import android.content.Intent;
 import android.os.Bundle;
+import android.os.SystemClock;
 import android.support.annotation.Nullable;
-import android.support.design.widget.Snackbar;
 import android.support.v7.app.AppCompatActivity;
 import android.support.v7.widget.GridLayoutManager;
 import android.support.v7.widget.RecyclerView;
 import android.support.v7.widget.Toolbar;
-import android.view.View;
 import android.view.Menu;
 import android.view.MenuItem;
 
@@ -47,7 +49,33 @@
         ExamAdapter adapter = new ExamAdapter(this, listOfExam.getListOfExam());
         recyclerView.setAdapter(adapter);
 
+        //zaczynamy powiadomienia
+
+        startAlarm(true,true);
+
     }
+
+    private void startAlarm(boolean isNotification, boolean isRepeat) {
+        AlarmManager manager = (AlarmManager)getSystemService(Context.ALARM_SERVICE);
+        Intent myIntent;
+        PendingIntent pendingIntent;
+
+        // SET TIME HERE
+        Calendar calendar= Calendar.getInstance();
+        //calendar.set(Calendar.HOUR_OF_DAY,22);
+        //calendar.set(Calendar.MINUTE,40);
+
+
+        myIntent = new Intent(MainActivity.this, AlarmNotificationReceiver.class);
+        pendingIntent = PendingIntent.getBroadcast(this,0,myIntent,0);
+
+
+        if(!isRepeat)
+            manager.set(AlarmManager.RTC_WAKEUP, SystemClock.elapsedRealtime(),pendingIntent);
+        else
+            manager.setRepeating(AlarmManager.RTC_WAKEUP, calendar.getTimeInMillis(), AlarmManager.INTERVAL_FIFTEEN_MINUTES/30,pendingIntent);
+    }
+
 
     @Override
     protected void onResume() {
Index: app/src/main/java/hanas/dnidomatury/MyService.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/hanas/dnidomatury/MyService.java	(date 1535490033632)
+++ app/src/main/java/hanas/dnidomatury/MyService.java	(date 1535490033632)
@@ -0,0 +1,55 @@
+package hanas.dnidomatury;
+
+import android.app.AlarmManager;
+import android.app.PendingIntent;
+import android.app.Service;
+import android.content.Context;
+import android.content.Intent;
+import android.os.IBinder;
+import android.os.SystemClock;
+
+import java.util.Calendar;
+import java.util.TimeZone;
+
+
+public class MyService extends Service {
+    public MyService() {
+    }
+
+    @Override
+    public IBinder onBind(Intent intent) {
+        return null;
+    }
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        startAlarm(true,true);
+    }
+
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        return START_NOT_STICKY;
+    }
+
+    private void startAlarm(boolean isNotification, boolean isRepeat) {
+        AlarmManager manager = (AlarmManager)getSystemService(Context.ALARM_SERVICE);
+        Intent myIntent;
+        PendingIntent pendingIntent;
+
+        //THIS IS WHERE YOU SET NOTIFICATION TIME FOR CASES WHEN THE NOTIFICATION NEEDS TO BE RESCHEDULED
+        Calendar calendar= Calendar.getInstance();
+        calendar.setTimeZone(TimeZone.getTimeZone("Europe/Warsaw"));
+        calendar.set(Calendar.HOUR_OF_DAY,23);
+        calendar.set(Calendar.MINUTE,0);
+
+        myIntent = new Intent(this,AlarmNotificationReceiver.class);
+        pendingIntent = PendingIntent.getBroadcast(this,0,myIntent,0);
+
+
+        if(!isRepeat)
+            manager.set(AlarmManager.RTC_WAKEUP, SystemClock.elapsedRealtime()+3000,pendingIntent);
+        else
+            manager.setRepeating(AlarmManager.RTC_WAKEUP, calendar.getTimeInMillis(), AlarmManager.INTERVAL_FIFTEEN_MINUTES ,pendingIntent);
+    }
+}
\ No newline at end of file
Index: app/src/main/java/hanas/dnidomatury/SelectActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/hanas/dnidomatury/SelectActivity.java	(revision 7332953f153c5c921c6b97532e8141a8e14b3ad5)
+++ app/src/main/java/hanas/dnidomatury/SelectActivity.java	(date 1535487073070)
@@ -46,7 +46,7 @@
         FullListExamAdapter adapter = new FullListExamAdapter(this, listOfExam.getListOfExam());
         recyclerView.setAdapter(adapter);
 
-        //ustawianie alarmu
+        /*//ustawianie alarmu
         alarmMgr = (AlarmManager) getSystemService(Context.ALARM_SERVICE);
         Intent intent = new Intent(this, AlarmReceiver.class);
         alarmIntent = PendingIntent.getBroadcast(this, 0, intent, 0);
@@ -57,7 +57,7 @@
         calendar.setTimeZone(TimeZone.getTimeZone("Europe/Warsaw"));
         calendar.set(Calendar.HOUR_OF_DAY, 21);
         calendar.set(Calendar.MINUTE, 22);
+        alarmMgr.setInexactRepeating(AlarmManager.RTC_WAKEUP, Calendar.getInstance().getTimeInMillis(), AlarmManager.INTERVAL_FIFTEEN_MINUTES / 15, alarmIntent);
         NotificationCompat.Builder mBuilder = new NotificationCompat.Builder(this, "channel1")
                 .setSmallIcon(R.drawable.ic_add)
                 .setContentTitle("My notification")
@@ -73,7 +73,7 @@
         notificationManager.notify(132, mBuilder.build());
 
 
-
+*/
     }
 
     @Override
@@ -108,7 +108,7 @@
         return super.onOptionsItemSelected(item);
     }
 
-    private void createNotificationChannel() {
+    /*private void createNotificationChannel() {
         // Create the NotificationChannel, but only on API 26+ because
         // the NotificationChannel class is new and not in the support library
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
@@ -122,6 +122,6 @@
             NotificationManager notificationManager = getSystemService(NotificationManager.class);
             notificationManager.createNotificationChannel(channel);
         }
-    }
+    }*/
 
 }
Index: app/src/main/AndroidManifest.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/AndroidManifest.xml	(revision 7332953f153c5c921c6b97532e8141a8e14b3ad5)
+++ app/src/main/AndroidManifest.xml	(date 1535488336631)
@@ -30,6 +30,23 @@
             android:name=".AddTaskActivity"
             android:label="@string/activity_add_task_name"
             android:theme="@style/AppTheme.UserDialog" />
+
+        <receiver android:name=".BootReceiver"
+            android:enabled="true"
+            android:exported="true">
+            <intent-filter>
+                <action android:name="android.intent.action.BOOT_COMPLETED" />
+
+                <category android:name="android.intent.category.HOME" />
+            </intent-filter>
+        </receiver>
+        <receiver android:name=".AlarmNotificationReceiver"/>
+
+        <service
+            android:name=".MyService"
+            android:enabled="true"
+            android:exported="true"></service>
+
     </application>
 
 </manifest>
\ No newline at end of file
